File: Platform/codesignal/Intro/27.variableName/solution.js
function solution(name) {
  if (!isNaN(name[0])) return false;
  return name.split("").map(x => {
    if (x === '_') return true;
    if (x >= '0' && x <= '9') return true;
    if (x.toLowerCase() >= 'a' && x.toLowerCase() <= 'z') return true;
    return false;
  }).every(x => x);
}

function solution(name) {
  if(!isNaN(name[0])) return false;
  if(name.length != name.match(/[a-zA-Z_\d]+/)[0].length) return false;
  return true;
}

function solution(name) {
  return /^[a-z_]\w*$/i.test(name)
}


console.log(solution("var_1__Int"));

File: Platform/codesignal/Intro/27.variableName/solution.go
package main

import "fmt"

func solution(name string) bool {
	if name[0] >= '0' && name[0] <= '9' {
		return false
	}
	for _, x := range name {
		if !((x >= '0' && x <= '9') || (x >= 'a' && x <= 'z') || (x >= 'A' && x <= 'Z') || x == '_') {
			return false
		}
	}
	return true
}

func main() {
	name := "var_1__Int"
	fmt.Println(solution(name))
}
